packages/notifier/tools/subscribe.js

```diff
@@ -0,0 +1,148 @@
+import { E, Far } from '@endo/far';
+
+import '../src/types-ambient.js';
+
+const sink = () => {};
+
+/**
+ * Create a near iterable that corresponds to a potentially far one.
+ *
+ * @template T
+ * @param {ERef<AsyncIterable<T>>} itP
+ * @returns {AsyncIterable<T>}
+ */
+export const subscribe = itP =>
+  Far('AsyncIterable', {
+    [Symbol.asyncIterator]: () => {
+      const it = E(itP)[Symbol.asyncIterator]();
+      return Far('AsyncIterator', {
+        next: async () => E(it).next(),
+      });
+    },
+  });
+
+/**
+ * Asyncronously iterates over the contents of a PublicationList as they appear.
+ * This iteration must drop parts of publication records that are no longer
+ * needed so they can be garbage collected.
+ *
+ * @template T
+ * @param {PublicationList<T>} pubList
+ * @returns {AsyncIterableIterator<T, T>}
+ */
+const makeEachIterator = pubList => {
+  // To understand the implementation, start with
+  // https://web.archive.org/web/20160404122250/http://wiki.ecmascript.org/doku.php?id=strawman:concurrency#infinite_queue
+  return Far('EachIterator', {
+    next: () => {
+      const resultP = E.get(pubList).head;
+      pubList = E.get(pubList).tail;
+      // We expect the tail to be the "cannot read past end" error at the end
+      // of the happy path.
+      // Since we are wrapping that error with eventual send, we sink the
+      // rejection here too so it doesn't become an invalid unhandled rejection
+      // later.
+      void E.when(pubList, sink, sink);
+      return resultP;
+    },
+    [Symbol.asyncIterator]: () => makeEachIterator(pubList),
```

Comment:
Normal Iterators return theirselves from this method rather than constructing a new iterator (which is the proper province of a _non_-Iterator Iterable). Does this unusual behavior provide enough value to justify inclusion? And even if so, can it be provided in a less surprising way?

Reply:
It used to be provided by adding a `subscribe` method on the iterators.  I changed that to conform to AsyncIterableIterator because it was really annoying to do the typecasting necessary to actually use it as intended in client code.  I didn't know that would be unidiomatic.

@erights introduced this feature for the Carol use-case, though I found no uses of it in the SDK or in dapps.  I will defer to him as to the preferred path forward.

Reply:
Does the "Carol use-case" in question refer to the 

> Carol's code is like Bob's except lower level, using the `AsyncIterableIterator` interface directly.

example?

Reply:
> Does this unusual behavior provide enough value to justify inclusion? 

Yes. Or at least: yes, eventually. If no one is using this functionality now, we could certainly postpone it.

> And even if so, can it be provided in a less surprising way?

If we are thinking of postponing this functionality, we could do so cleanly by making the EachIterator just an Iterator for now, and postpone making it also an Iterable. I'd be happy with that, given that we leave a comment behind in the code explaining the possibility and the controversy. Linking to this conversation would be adequate.

Reply:
I'd prefer not to rip out all the Carol code.  Would `iterator.fork()` be an appropriate replacement for what I tried to name `iterator[Symbol.asyncIterator]()`?

Reply:
`fork` is perfect.

Reply:
> Normal Iterators return theirselves from this method rather than constructing a new iterator

The current behaviour is documented in the `README.md` as:

> Each produced `AsyncIterableIterator` is an `AsyncIterator`,
and is also an `AsyncIterable` with a `Symbol.asyncIterator` method of its own that when called
produces a new `AsyncIterableIterator`
whose starting point is the current position of its parent `AsyncIterableIterator`.

Reply:
BTW, in private chat, @erights added:
> I prefer `fork()` but am ok with `clone()`.